memcached.c

```diff
@@ -1640,13 +1640,7 @@ static void process_bin_get_or_touch(conn *c) {
         rsp->message.header.response.cas = htonll(ITEM_get_cas(it));
 
         // add the flags
-        if (settings.inline_ascii_response) {
-            rsp->message.body.flags = htonl(strtoul(ITEM_suffix(it), NULL, 10));
-        } else if (it->nsuffix > 0) {
-            rsp->message.body.flags = htonl(*((uint32_t *)ITEM_suffix(it)));
-        } else {
-            rsp->message.body.flags = 0;
-        }
+        rsp->message.body.flags = htonl(FLAGS_CONV(settings.inline_ascii_response, it));
```

Comment:
But over here, we may take further dispose for flag, with htonl util.
Should we use this solution
```c
FLAGS_CONV(settings.inline_ascii_response, it, rsp->message.body.flags);
rsp->message.body.flags = htonl(rsp->message.body.flags);
```
instead of
```c
rsp->message.body.flags = htonl(FLAGS_CONV(settings.inline_ascii_response, it));
```
to adjust the flag value after updated FLAGS_CONV routine.
Or just use extension?

Reply:
Yup, that's fine. Or use an explicitly named temporary variable. GCC will likely optimize it away.

I'd much prefer that than doing something forbidden by the C standard :)